### DISCUSSION TITLE
IS THE MINIO MIGRATION TO RELEASE.2022-10-29 ISSUE SOLVED?

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#9940] [Help]
Hi I am using Self Host Budibase deployed using docker compose. From time to time I check for docker compose changes and I pull latest images for all components. Since data is stored in docker volumes I have never faced issues so far, till last October when new MinIO was released.

If you had set a Budibase before [RELEASE.2022-10-29T06-21-33Z] of MinIO it was using a Filesystem that is no longer supported. So when you try to create a new Stand alone Budibase with docker-compose keeping the docker volumes so that you don't loose your existing work MinIO now gives an error and applications are not accessible of course.
Currently the only solution is to freeze MinIO version to [RELEASE.2022-10-24T18-35-07Z] in docker compose and continue using latest for other images.
Relevant documentation: https://min.io/docs/minio/linux/operations/install-deploy-manage/migrate-fs-gateway.html
And profoundly this is not an easy to follow guide to migrate.

What MinIO version is used in the Stand alone docker image?

And the main question is , should someone pull latest docker images from time to time for all docker-compose modules? Could for a example a new couch db version be incompatible with the data in its docker volume at some point? Or someone should not do that and update only from inside the Budibase (which in fact updates only the 3 Budibase core images to the latest) and never touch again the other images. But i guess it that was the case you would use fixed versions and not latest. 

Please advice

**Comment by shogunpurple** [comment]
Hi @gmentzik - you are correct, we had actually made some changes to the `docker-compose.yaml` to pin to the old version of MinIO, but only in dev and it appears it must have been overlooked in the production `docker-compose.yaml` and single image config. I've updated both the single image download script and the production docker compose configuration to use the pinned version, and once merged I'll roll out a release with them both.

From there, you shouldn't need to do the migration as the version will be pinned. New couchDB versions should be fine, but this was just a case of an oversight that we should have pinned the minIO version.

Follow along with the PR here:

https://github.com/Budibase/budibase/pull/9943

Thanks for raising!

**Reply by gmentzik** [reply]
No prob. Thank you for your prompt response. I just wonder if I tried a cli backup and restore to a fresh Budibase installation would it work for MinIO or the issue will still remain? Saying that, I haven't tried a cli backup recently as past experience was not so stable for me so I periodically backup docker images and volumes instead, which works fine for me.

**Reply by shogunpurple** [reply]
@gmentzik we just had a discussion around this internally actually, and realised that if we do this, it causes the opposite of what you are seeing - it breaks **newer** installations after `RELEASE.2022-10-24T18-35-07Z` as it will install the old version of the `minio` image after they have already been using the new filesystem.

In this case, what you have done there already is fine, and you will need to keep the frozen MinIO image tag in your `docker-compose.yaml` to maintain backwards compatibility.

We are making some changes to the budibase CLI so that it will not replace this if you have already updated it - so you can keep using the CLI to upgrade in future.

**Reply by melohagan** [reply]
Docs: https://docs.budibase.com/docs/budibase-cli-setup

**Reply by gmentzik** [reply]
@shogunpurple I understand the problem and I thought too that latest MinIO would have issues going backwards. 
But please tell me if I use cli-backup from old a Budibase with old MinIO, will  restore to a new Budibase installation with new MinIO  work?
In any case freezing MinIO to an old version forever is not a valid scenario for long.

**Reply by gmentzik** [reply]
Question is still open. Question is: How to migrate/update to a Self host Budibase  with a newer MinIO version than RELEASE.2022-10-24T18-35-07Z?

**Reply by mike12345567** [reply]
@gmentzik The best option would be to migrate your apps, or do a full installation migration via the CLI - this will allow exporting and then re-importing to a new version of Budibase.

**Reply by gmentzik** [reply]
@mike12345567 
migrate your apps = export/import app? Yes but any files at Minio need to reloaded  manually this way. 
full installation migration via the CLI = backup/ restore with budi cli?
Please clarify.

**Reply by mike12345567** [reply]
Hi @gmentzik - with the latest version of Budibase files in MinIO are exporting as part of the app exports, so there is no need for a manual reload of files.

A full installation migration would mean using a backup created by the CLI and then importing it to a fresh Budibase install.

**Reply by gmentzik** [reply]
FYI I created a new Budibase host with cli tool which means when budinase started it fetched the latest MinIO. Then restored Backup I had taken with CLI before at old host and applications work fine. I don't understand how is this possible as you just backup the minIO data folder, or am I missing something. So backup and restore to fresh installation seams to work. I will run further test and come back.

**Reply by mike12345567** [reply]
Hi @gmentzik - we don't actually perform backups at the docker level, we do some work at the application level to transfer the data out, this means that you can change the underlying configuration of your installation (for example, move from docker-compose to single image, to kubernetes etc) and the backup and restore will still work.

**Reply by gmentzik** [reply]
> Hi @gmentzik - we don't actually perform backups at the docker level, we do some work at the application level to transfer the data out, this means that you can change the underlying configuration of your installation (for example, move from docker-compose to single image, to kubernetes etc) and the backup and restore will still work.

Is the cli backup and restore source code available to understand the commands you execute to backup/restore minio and couchdb? This would help me understand how to keep a safe backup. Right now I am backing up docker images and volumes but it seams that cli can perform a safe backup-restore procedure too. When I had tested cli backup/restore some months ago it had issues, so went for alternative backup procedure.

**Reply by mike12345567** [reply]
Hi @gmentzik - the source code for the CLI backups can be found here: https://github.com/Budibase/budibase/tree/develop/packages/cli/src/backups

**Reply by gmentzik** [reply]
> Hi @gmentzik - the source code for the CLI backups can be found here: https://github.com/Budibase/budibase/tree/develop/packages/cli/src/backups

Had a quick look. It seams to me that you have found a way to backup and restore MinIO buckets contents provided they exists at new environment and that's why it works for you. I wonder why MinIO guys don't provide this information.
